controllers/secret_controller_test.go

```diff
@@ -0,0 +1,160 @@
+package controllers
+
+import (
+	"testing"
+
+	"github.com/3scale-labs/authorino/api/v1beta1"
+	"gotest.tools/assert"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type FakeReconciler struct {
+	Reconciled bool
+}
+
+func (r *FakeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	r.Reconciled = true
+	return ctrl.Result{}, nil
+}
+
+func (r *FakeReconciler) Reset() {
+	r.Reconciled = false
+}
+
+func TestReconciler(t *testing.T) {
```

Comment:
Maybe we could do the creation of Service and Secrets outside the `TestReconciler` func, and separate the assertions addressing the type of secrets and if the service was reconciled or not? That way it would be easier to debug and to keep track of coverage

Reply:
I ended up creating a `<Subject>Test` struct. Our former approach of declaring variables that are reused across multiple tests in the same test file easily run into naming conflicts with other files of the same package.